ios/FluentUI/Avatar/Avatar.swift

```diff
@@ -100,46 +108,63 @@ public struct Avatar: View, TokenizedControlView {
-        let shouldUseCalculatedColors = !initialsString.isEmpty && style != .overflow
 
         // Adding ringInnerGapOffset to ringInnerGap & ringThickness to accommodate for a small space between
         // the ring and avatar when the ring is visible and there is no inner ring gap
         let ringInnerGapOffset = 0.5
         let ringInnerGap: CGFloat = isRingVisible ? (hasRingInnerGap ? tokenSet[.ringInnerGap].float : -ringInnerGapOffset) : 0
         let ringThickness: CGFloat = isRingVisible ? (hasRingInnerGap ? ringThicknessToken : ringThicknessToken + ringInnerGapOffset) : 0
         let ringOuterGap: CGFloat = isRingVisible ? tokenSet[.ringOuterGap].float : 0
+        let totalRingGap: CGFloat = ringInnerGap + ringThickness + ringOuterGap
         let avatarImageSize: CGFloat = contentSize
         let ringInnerGapSize: CGFloat = avatarImageSize + (ringInnerGap * 2)
         let ringSize: CGFloat = ringInnerGapSize + (ringThickness * 2)
         let ringOuterGapSize: CGFloat = ringSize + (ringOuterGap * 2)
-        let presenceIconSize: CGFloat = AvatarTokenSet.presenceIconSize(state.size)
-        let presenceIconOutlineSize: CGFloat = presenceIconSize + (tokenSet[.presenceIconOutlineThickness].float * 2)
-
-        // Calculates the positioning of the presence icon ensuring its center is always on top of the avatar circle's edge
-        let ringInnerGapRadius: CGFloat = (ringInnerGapSize / 2)
-        let ringInnerGapHypotenuse: CGFloat = sqrt(2 * pow(ringInnerGapRadius, 2))
-        let presenceIconHypotenuse: CGFloat = sqrt(2 * pow(presenceIconOutlineSize / 2, 2))
-        let presenceFrameHypotenuse: CGFloat = ringInnerGapHypotenuse + ringInnerGapRadius + presenceIconHypotenuse
-        let presenceIconFrameSideRelativeToInnerRing: CGFloat = sqrt(pow(presenceFrameHypotenuse, 2) / 2)
-
-        // Creates positioning coordinates for the presence cutout (enabling the transparency of the presence icon)
-        let outerGapAndRingThicknesCombined: CGFloat = ringOuterGap + ringThickness
-        let presenceIconFrameDiffRelativeToOuterRing: CGFloat = ringOuterGapSize - (presenceIconFrameSideRelativeToInnerRing + outerGapAndRingThicknesCombined)
-        let presenceCutoutOriginXLTR = ringOuterGapSize - presenceIconFrameDiffRelativeToOuterRing - presenceIconOutlineSize
-        let presenceCutoutOriginXRTL = presenceIconFrameDiffRelativeToOuterRing
-        let presenceCutoutOriginX: CGFloat = layoutDirection == .rightToLeft ? presenceCutoutOriginXRTL : presenceCutoutOriginXLTR
-        let presenceCutoutOriginY = presenceCutoutOriginXLTR
-        let presenceIconFrameSideRelativeToOuterRing: CGFloat = presenceIconFrameSideRelativeToInnerRing + outerGapAndRingThicknesCombined
-        let overallFrameSide = max(ringOuterGapSize, presenceIconFrameSideRelativeToOuterRing)
+
+        // Avatar accessory calculation
+        let accessoryImage: Image = {
```

Comment:
dumb question, we can't just call Image(uiImage: state.activityImage) in line 308 not remove this accessoryImage variable all together?

Reply:
state.activityImage is an optional and we can't pass a nil image in SwiftUI

Reply:
We can't make this non-optional unless we want to provide a default image, which was agreed that we shouldn't in above threads https://github.com/microsoft/fluentui-apple/pull/1502#discussion_r1082908905